mqtt/init.lua

```diff
@@ -81,6 +81,183 @@ function mqtt.run_sync(cl)
 	end
 end
 
+
+--- Validates a topic with wildcards.
+-- @param t (string) wildcard topic to validate
+-- @return topic, or false+error
+function mqtt.validate_subscribe_topic(t)
+	if type(t) ~= "string" then
```

Comment:
please follow such errors checking behavior: https://github.com/xHasKx/luamqtt/blob/master/mqtt/init.lua#L10

Reply:
since it is validation, the output of a validation function should be usable as input for an assertion. Hence this function does not throw errors itself, and returns the input value if valid.

This is exactly what I implemented for the other functions, as they should throw errors, as documented.